frame/alliance/src/lib.rs

```diff
@@ -705,6 +711,15 @@ pub mod pallet {
 		pub fn join_alliance(origin: OriginFor<T>) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 
+			// We don't want anyone to join as an Ally before the Founders/Fellows have been
```

Comment:
Minor note, 
Here the logic of what means is_initialized is encapsulated in the method.
But the comments describes the logic within the is_initialized function. 
This might lead a comment to be outdated and wrong, if logic within the function is_initialized changes. 

For such a cases it is better to be more abstract, and write something like 
`We don't want anyone to join as an Ally before Alliance is initialized via Root call`.

The function itself can have a doc describing the functionality within it, if someone changes the function behaviour,  there is less chance the doc will be forgotten to be changed.